--- /var/www/openmediavault/js/omv/module/admin/FileSystemMgmt.js	2013-02-16 21:08:03.000000000 -0500
+++ /var/www/openmediavault.devel/js/omv/module/admin/FileSystemMgmt.js	2013-03-24 20:28:49.973249695 -0400
@@ -382,7 +382,7 @@
 			} else {
 				OMV.MessageBox.error(null, error);
 			}
-		}, this, "FileSystemMgmt", "mount", { "id": record.get("uuid"),
+		}, this, "FileSystemMgmt", "mount", { "id": record.get("uuid"),"devicefile": record.get("devicefile"),
 		"fstab": true });
 	},
 
