--- /usr/share/php/openmediavault/system.inc	2013-02-16 21:08:03.000000000 -0500
+++ /devel/usr/share/php/openmediavault/system.inc	2013-03-16 17:13:00.000000000 -0400
@@ -23,6 +23,7 @@
 require_once("openmediavault/object.inc");
 require_once("openmediavault/functions.inc");
 require_once("openmediavault/util.inc");
+require_once("openmediavault/loop.inc");
 
 /**
  * @class OMVSystem
@@ -492,6 +493,11 @@
 		$this->ptType = $data['pttype'];
 		$this->usage = $data['usage'];
 
+		// fix filesystems that don't have UUID's iso9660???
+		if(!$this->uuid && $this->type === "iso9660") {
+			$this->uuid=$this->deviceFile;
+		}
+
 		// Set flag to mark informations has been successfully read.
 		$this->dataCached = TRUE;
 
@@ -882,6 +888,7 @@
 	protected $model = FALSE;
 	protected $vendor = FALSE;
 	protected $udevInfo = FALSE;
+	protected $serial = FALSE;
 
 	/**
 	 * Constructor
@@ -973,12 +980,19 @@
 	 */
 	public function getModel() {
 		if($this->model === FALSE) {
-			$filename = sprintf("/sys/block/%s/device/model",
-			  $this->getDeviceName());
-			if(file_exists($filename)) {
-				$this->model = trim(file_get_contents($filename));
+			if(!strncmp($this->getDeviceName(),"loop",4)) {
+				$loop = new OMVLoopDevice($this->deviceFile);
+				$this->model=$loop->getName();
+				$this->vendor="Loop Device";
+				$this->serial=$loop->getSerial();
 			} else {
-				$this->model = "";
+				$filename = sprintf("/sys/block/%s/device/model",
+				  $this->getDeviceName());
+				if(file_exists($filename)) {
+					$this->model = trim(file_get_contents($filename));
+				} else {
+					$this->model = "";
+				}
 			}
 		}
 		return $this->model;
@@ -990,12 +1004,16 @@
 	 */
 	public function getVendor() {
 		if($this->vendor === FALSE) {
-			$filename = sprintf("/sys/block/%s/device/vendor",
-			  $this->getDeviceName());
-			if(file_exists($filename)) {
-				$this->vendor = trim(file_get_contents($filename));
+			if(!strncmp($this->getDeviceName(),"loop",4)) {
+				$this->vendor = "Loop Device";
 			} else {
-				$this->vendor = "";
+				$filename = sprintf("/sys/block/%s/device/vendor",
+				  $this->getDeviceName());
+				if(file_exists($filename)) {
+					$this->vendor = trim(file_get_contents($filename));
+				} else {
+					$this->vendor = "";
+				}
 			}
 		}
 		return $this->vendor;
@@ -1006,9 +1024,12 @@
 	 * @return The device serial number.
 	 */
 	public function getSerialNumber() {
-		if(FALSE === ($property = $this->getUdevProperty("ID_SERIAL_SHORT")))
-			return "n/a";
-		return str_replace("_", " ", $property);
+		if($this->serial === FALSE) {
+			if(FALSE === ($property = $this->getUdevProperty("ID_SERIAL_SHORT")))
+				return "n/a";
+			return str_replace("_", " ", $property);
+		}
+		return $this->serial;
 	}
 
 	/**
